### DISCUSSION TITLE
Top-level statements namespace keyword support

### DISCUSSION THREAD

**Discussion by sefacan** [dotnet/csharplang#4348] [Language Ideas]
Hi there,
I like top-level statements in csharp 9 and also I want to use namespace without curly braces.

```csharp
namespace ConsoleApp;

using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

Console.WriteLine("Hello world!");
```

or can be
```csharp
namespace ConsoleApp;

using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

public static void Main(string[] args)
{
    Console.WriteLine("Hello world!");
}
```

can be also
```csharp
namespace ConsoleApp;

using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

public class Program
{
    public static void Main(string[] args)
    {
        Console.WriteLine("Hello world!");
    }
}
```

I only want to discuss about of this idea.

**Comment by YairHalberstadt** [comment]
This is currently championed for C# 10. See the meeting notes for a discussion about using file scoped namespaces in conjunction with top level statements. You can find them linked in [this discussion](https://github.com/dotnet/csharplang/discussions/4306).

**Comment by sefacan** [comment]
Thank you @YairHalberstadt it will be great at least for me.
I have seen some people think its unnecessary thing for spending much time to this topic.
